src/repo/mod.rs

```diff
@@ -98,126 +86,114 @@ impl<TRepoTypes: RepoTypes> Repo<TRepoTypes> {
         datastore_path.push("datastore");
         let block_store = TRepoTypes::TBlockStore::new(blockstore_path);
         let data_store = TRepoTypes::TDataStore::new(datastore_path);
-        let (sender, receiver) = channel::<RepoEvent>();
+        let (sender, receiver) = channel::<RepoEvent>(1);
         (Repo {
             block_store,
             data_store,
             events: sender,
         }, receiver)
     }
 
-    pub fn init(&self) -> impl Future<Output=Result<(), Error>> {
+    pub async fn init(&self) -> Result<(), Error> {
         let block_store = self.block_store.clone();
         let data_store = self.data_store.clone();
-        async move {
-            let f1 = block_store.init();
-            let f2 = data_store.init();
-            let (r1, r2) = join!(f1, f2);
-            if r1.is_err() {
-                r1
-            } else {
-                r2
-            }
+        // FIXME: unsure why this fails with failure::core::future::future::Future
```

Comment:
Well no longer it doesn't. Not sure if `failure` re-exports something unexpected since it appears where there should be only `core::future::Future`. Might be a bug in nightly.